### DISCUSSION TITLE
How to control the depth to which the prettyfied json is produced?

### DISCUSSION THREAD

**Discussion by WurmD** [nlohmann/json#2648] [Ideas]
How to control the depth to which the prettyfied json is produced?

For the following json

`{"data": [{"frame_index": 1, "skeleton": [{"pose": [0.612, 0.639, 0.624]}]}]}`

The current prettyfying yields

```
{
    "data": [
        {
            "frame_index": 1,
            "skeleton": [
                {
                    "pose": [
                        0.612,
                        0.639,
                        0.624,
                     ]
                 }
             ]
         }
      ]
}
```

**Desired behavior:**
If depth=0 then the output would be equal to the non-prettyfied json

If depth=3 then the output would be

```
{
    "data": [
        {
            "frame_index": 1,
            "skeleton": [
                {
                    "pose": [ 0.612, 0.639, 0.624, ]
                 }
             ]
         }
      ]
}
```

Is this possible in nlohmann/json ?

**Comment by gregmarr** [comment]
There is not currently support for keeping some parts on the same line.  It has been discussed in the past.  Here's one example: 
https://github.com/nlohmann/json/pull/1121
